### DISCUSSION TITLE
Part I . 1 - ssl.SSLError: [SSL] record layer failure (_ssl.c:1020) on macOS

### DISCUSSION THREAD

**Discussion by thisisjaymehta** [browserengineering/book#1361] [Q&A]
I am encountering an issue in Part I where I cannot connect to a website using HTTPS.

Here is the error message:
`ssl.SSLError: [SSL] record layer failure (_ssl.c:1020)`

I followed the instructions provided in a [StackOverflow answer](https://stackoverflow.com/questions/52805115/certificate-verify-failed-unable-to-get-local-issuer-certificate) to install certificates. However, I couldn't find the `Install Certificates.command` file in my Python virtual environment. Additionally, there is no Python directory in my Applications folder. Although Python 3 is preinstalled on my system, it doesn’t seem to have the necessary directory in the Applications folder.

I tried running `pip install --upgrade certifi`, but the error persists. My error also differs from the one mentioned in the linked StackOverflow question.

I’ve searched for solutions specific to my error, but they all suggest installing certificates, which I cannot locate in my Python installation.

Any help in resolving this issue would be greatly appreciated. Thank you for your time!

**Comment by pavpanchekha** [comment]
To start with, what OS are you on? Are you on macOS? Which version?

**Reply by thisisjaymehta** [reply]
Yes, macos sequoia 15.2

**Reply by pavpanchekha** [reply]
Apologies for totally losing that this thread existed, but have you tried installing the homebrew package `python-certifi`, which I think in turn will install the Homebrew package `ca-certificates`? I believe this is what my system uses.
